tools/slack-release-notifier/src/main.ts

```diff
@@ -0,0 +1,118 @@
+/**
+ * @license
+ * Copyright 2019 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { getInput } from '@actions/core';
+import { context, GitHub } from '@actions/github';
+import { IncomingWebhook } from '@slack/webhook';
+import { ReposGetReleaseByTagResponse } from '@octokit/rest';
+
+import { getReleaseType, getTextForReleaseType } from './utils/release-type';
+import { convertToMrkdwn } from './utils/convert-to-mrkdwn';
+import { findPreviousRelease } from './utils/find-previous-release';
+
+function getReleaseTitle(release: ReposGetReleaseByTagResponse) {
+  if (!release.name) {
+    return release.tag_name;
+  }
+
+  return release.name.includes(release.tag_name)
+    ? release.name
+    : `${release.name} (${release.tag_name})`;
+}
+
+async function run(): Promise<void> {
+  const slackWebhookUrl = process.env.SLACK_WEBHOOK_URL;
+  const githubToken = getInput('repo-token', { required: true });
+
+  if (!slackWebhookUrl)
+    throw new Error('Required environment variables are not set.');
```

Comment:
NIT: I'd add that the slack webhook env variable is missing instead of a generic error message.